tests/testthat/test-step_date_hcai.R

```diff
@@ -0,0 +1,142 @@
+  d_hcai <- recipes::bake(date_rec, newdata = d)
+
+  expect_equal(d_hcai$f_DTS_hour, expected$f_DTS_hour)
+})
+
+test_that("continuous - check date time column created correctly", {
+  expected <- data.frame(
+    b_nums = d$b_nums,
+    f_DTS_dow_sin = sin(2 * pi / 7 * c(7, 2, 5)),
+    f_DTS_dow_cos = cos(2 * pi / 7 * c(7, 2, 5)),
+    f_DTS_month_sin = sin(2 * pi / 12 * c(1, 1, 5)),
+    f_DTS_month_cos = cos(2 * pi / 12 * c(1, 1, 5)),
+    f_DTS_year = c(2000, 2000, 2001),
+    f_DTS_hour_sin = sin(2 * pi / 24 * c(13, 1, 12)),
+    f_DTS_hour_cos = cos(2 * pi / 24 * c(13, 1, 12))
+  )
+
+  d <- d %>% dplyr::select(b_nums, f_DTS)
+  cols <- find_date_cols(d)
+
+  date_rec <- recipes::recipe(head(d), ~ .)
+  date_rec <- step_date_hcai(date_rec, cols = cols, feature_type = "continuous") %>%
+    recipes::step_rm(cols = cols)
+  date_rec <- recipes::prep(date_rec, training = d)
+  d_hcai <- recipes::bake(date_rec, newdata = d)
+
+  expect_equal(d_hcai, expected)
+})
+
+test_that("continuous - check date only column created correctly - no hour", {
+  expected <- data.frame(
+    b_nums = d$b_nums,
+    c_DTS_dow_sin = sin(2 * pi / 7 * c(2, 2, 4)),
+    c_DTS_dow_cos = cos(2 * pi / 7 * c(2, 2, 4)),
+    c_DTS_month_sin = sin(2 * pi / 12 * c(3, 3, 6)),
+    c_DTS_month_cos = cos(2 * pi / 12 * c(3, 3, 6)),
+    c_DTS_year = c(2018, 2018, 2019)
+  )
+
+  d <- d %>% dplyr::select(b_nums, c_DTS)
+  cols <- find_date_cols(d)
+
+  date_rec <- recipes::recipe(head(d), ~ .)
+  date_rec <- step_date_hcai(date_rec, cols = cols, feature_type = "continuous") %>%
+    recipes::step_rm(cols = cols)
+  date_rec <- recipes::prep(date_rec, training = d)
+  d_hcai <- recipes::bake(date_rec, newdata = d)
+
+  expect_equal(data.frame(d_hcai), expected)
+})
+
+test_that("Print method works correctly", {
+  d <- d %>% dplyr::select(e_date, b_nums, d_chars)
+  cols <- find_date_cols(d)
+  sdf <- c("dow", "month", "year")
+
+  date_rec <- recipes::recipe(head(d), ~ .)
+  date_rec <- step_date_hcai(date_rec, cols = cols, feature_type = "categories")
+  date_rec <- recipes::prep(date_rec, training = d)
+
+  expect_output(
+    print(date_rec),
+    regexp = "Date features from e_date"
+  )
+})
+
+test_that("tidy method prints correctly for categories features", {
+  d <- d %>% dplyr::select(e_date, b_nums, d_chars)
+  cols <- find_date_cols(d)
+
+  date_rec <-
+    recipes::recipe(head(d), ~ .) %>%
+    step_date_hcai(cols = cols, feature_type = "categories") %>%
+    recipes::prep(training = d)
+
+  exp <- tibble::as_tibble(
+    data.frame(
+      terms = "e_date",
+      feature_type = "categories"
+    )
+  )
+
+  expect_equal(
+    exp,
+    broom::tidy(date_rec$steps[[1]]))
+  expect_s3_class(broom::tidy(date_rec$steps[[1]]), "tbl_df")
+})
+
+test_that("test possible values for `feature_type`", {
+  expect_error(
+    recipes::recipe(head(d), ~ .) %>%
+      step_date_hcai(cols = cols, feature_type = "test"),
+    "Possible values"
+  )
+})
```

Comment:
I agree. I didn't test it initially because `ord2fac` was copied from Max Kuhn's code before I added anything. I was copying the style. Since there are a lot of changes from his code in other functions, I think you are right that this is important to add. Good catch.